components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -519,15 +521,25 @@ private void validateAddOrganizationRequest(Organization organization) throws Or
     private void validateOrgNameUniquenessAmongSiblings(String parentOrgId, String organizationName)
             throws OrganizationManagementException {
 
-        boolean hasSiblingWithSameName =
-                organizationManagementDAO.getChildOrganizations(parentOrgId.trim(), false).stream().anyMatch(
-                        basicOrganization -> StringUtils.equals(basicOrganization.getName(), organizationName.trim()));
-        if (hasSiblingWithSameName) {
+        if (organizationManagementDAO.isSiblingOrganizationExistWithName(organizationName, parentOrgId)) {
             throw handleClientException(ERROR_CODE_SAME_ORG_NAME_ON_IMMEDIATE_SUB_ORGANIZATIONS_OF_PARENT_ORG,
                     organizationName, parentOrgId);
         }
     }
 
+    private void validateOrgNameUniqueness(String parentOrgId, String organizationName)
```

Comment:
Let's add a unit test covering this function's process

Reply:
Added unit tests by this commit - https://github.com/wso2-extensions/identity-organization-management-core/pull/65/commits/13e966198010e05a2e60d6a7cd7cc301d9ed0c9a#diff-96228f2c7b6a184e56f8c4f5b0ec5878c01dc60313cbf7ac79ee197005edf5b6